File: core/src/main/java/com/ustadmobile/core/catalog/contenttype/EPUBTypePlugin.java
package com.ustadmobile.core.catalog.contenttype;

import com.ustadmobile.core.view.EpubContentView;

import java.util.Arrays;
import java.util.List;

/**
 * Created by mike on 9/9/17.
 */

public class EPUBTypePlugin extends ContentTypePlugin {

    public static final String[] MIME_TYPES = new String[]{"application/epub+zip"};

    public static final String[] EXTENSIONS = new String[]{"epub"};

    public static final String OCF_CONTAINER_PATH = "META-INF/container.xml";

    @Override
    public String getViewName() {
        return EpubContentView.VIEW_NAME;
    }

    @Override
    public List<String> getMimeTypes() {
        return Arrays.asList(MIME_TYPES);
    }

    @Override
    public List<String> getFileExtensions() {
        return Arrays.asList(EXTENSIONS);
    }

}

File: core/src/main/java/com/ustadmobile/core/catalog/contenttype/EPUBTypePlugin.kt
package com.ustadmobile.core.catalog.contenttype

import com.ustadmobile.core.view.EpubContentView

import java.util.Arrays

/**
 * Created by mike on 9/9/17.
 */

class EPUBTypePlugin : ContentTypePlugin() {

    override val viewName: String
        get() = EpubContentView.VIEW_NAME

    override val mimeTypes: List<String>
        get() = Arrays.asList(*MIME_TYPES)

    override val fileExtensions: List<String>
        get() = Arrays.asList(*EXTENSIONS)

    companion object {

        val MIME_TYPES = arrayOf("application/epub+zip")

        val EXTENSIONS = arrayOf("epub")

        val OCF_CONTAINER_PATH = "META-INF/container.xml"
    }

}
